packages/@webex/plugin-cc/src/cc.ts

```diff
@@ -245,4 +257,54 @@ export default class ContactCenter extends WebexPlugin implements IContactCenter
       allowMultiLogin: this.$config?.allowMultiLogin ?? true,
     };
   }
+
+  /**
+   * Called when we reconnection has been completed
+   */
+  private async handleConnectionLost(msg: ConnectionLostDetails): Promise<void> {
+    this.webSocketManager.handleConnectionLost(msg);
+
+    if (msg.isConnectionLost) {
+      // TODO: Emit an event saying connection is lost
+      this.$webex.logger.info('event=handleConnectionLost | Connection lost');
+    } else if (msg.isSocketReconnected) {
+      // TODO: Emit an event saying connection is re-estabilished
+      this.$webex.logger.info(
+        'event=handleConnectionReconnect | Connection reconnected attempting to request silent relogin'
+      );
+      // TODO: Add a check here for config (if it has property to auto silent relogin) (https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-574666)
+      await this.silentRelogin();
+    }
+  }
+
+  /**
+   * Called when we finish registration to silently handle the errors
+   */
+  private async silentRelogin(): Promise<void> {
+    try {
+      const reLoginResponse = await this.services.agent.reload();
+      const {auxCodeId, agentId, lastStateChangeReason} = reLoginResponse.data;
+
+      if (lastStateChangeReason === 'agent-wss-disconnect') {
+        this.$webex.logger.info(
+          'event=requestAutoStateChange | Requesting state change to available on socket reconnect'
+        );
+        const stateChangeData: StateChange = {
+          state: AGENT_STATE_AVAILABLE,
+          auxCodeId,
+          lastStateChangeReason,
+          agentId,
+        };
+        await this.setAgentState(stateChangeData);
+      }
+      // TODO: Update login state of config here (https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-574666)
+    } catch (error) {
+      if (getErrorReason(error) === 'AGENT_NOT_FOUND') {
```

Comment:
Again, silently handle this fialure (mostly will occur on fresh reload if the user is not logged in).